src/internal_events/route.rs

```diff
@@ -3,11 +3,12 @@ use vector_core::internal_event::InternalEvent;
 
 #[derive(Debug)]
 pub struct RouteEventDiscarded<'a> {
-    pub output: &'a str,
+    pub outputs: Vec<&'a str>,
 }
 
 impl<'a> InternalEvent for RouteEventDiscarded<'a> {
     fn emit_metrics(&self) {
-        counter!("events_discarded_total", 1, "output" => self.output.to_string());
+        let outputs = self.outputs.join(", ");
+        counter!("events_discarded_total", self.outputs.len() as u64, "outputs" => outputs);
```

Comment:
I think this isn't going to be very useful as a metric tag. Maybe we should publish one metric per output?

Reply:
I agree with @lukesteensen maybe we don't really need this event. Would it be ok to drop it?

Reply:
I think if there will always be an `else` - it should have a `processed_events` for the `else` branch, as long as we document the change on the release notes (since it's breaking) I'm ok with that.